### ISSUE TITLE
Slicing some large documents containing large images does not reduce file size

### ISSUE THREAD

**Issue by ebousse** [junrrein/pdfslicer#225]
Consider this file: [example.pdf](https://github.com/junrrein/pdfslicer/files/7047692/example.pdf)

- Page 1 contains a rather large screenshot uncompressed (~2.4 MB)
- Page 2 contains a single line of text
- The size of the pdf is 3.2 MB, mostly due to the large image.

Now let's use PDF Slicer to remove the page with the large image and to only keep the page with text. The result is this file, **which is also 3.2 MB in size**, exactly as if the image was still part of the PDF (while it does not appear on any page!): [example_sliced.pdf](https://github.com/junrrein/pdfslicer/files/7047700/example_sliced.pdf)

Additionally, note that if we run `ps2pdf example_sliced.pdf example_sliced_ps2pdf.pdf` (which converts a PDF to PS and then again back), we obtain this file, which is only 4.7KB (as expected for a pdf with only one line of text): [example_sliced_ps2pdf.pdf](https://github.com/junrrein/pdfslicer/files/7047710/example_sliced_ps2pdf.pdf)

------

In short, it appears that PDF Slicer is not always able to actually "remove" from the output file an image that was part of a removed page?
